File: LeetCode Solutions/LeetCoding Challenge/June 2021 LeetCoding Challenge/Number of Subarrays with Bounded Maximum/Number of Subarrays with Bounded Maximum.py
class Solution:
    def numSubarrayBoundedMax(self, nums: List[int], left: int, right: int) -> int:
        numberOfSubarrays, lower, middle = 0, 0, 0
        for num in nums:
            if (num > right):
                middle = 0
            else:
                middle += 1
                numberOfSubarrays += middle
            if (num >= left):
                lower = 0
            else:
                lower += 1
                numberOfSubarrays -= lower
        return numberOfSubarrays

File: LeetCode Solutions/LeetCoding Challenge/June 2021 LeetCoding Challenge/Number of Subarrays with Bounded Maximum/Number of Subarrays with Bounded Maximum.cpp
class Solution {
public:
    int numSubarrayBoundedMax(vector<int>& nums, int left, int right) {
        int numberOfSubarrays = 0, lower = 0, middle = 0;
        for (int num:nums){
            if (num > right){
                middle = 0;
            } else{
                numberOfSubarrays += ++middle;
            }
            if (num >= left){
                lower = 0;
            } else{
                numberOfSubarrays -= ++lower;
            }
        }
        return numberOfSubarrays;
    }
};